### PR TITLE
Custom json matcher

### PR DESCRIPTION
**Pull request by tomaszpowroznik**

### CODE REVIEW

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/GeneratedOrderControllerSubmittedTest.java [hunk 1]
```diff
@@ -177,13 +169,13 @@ void submittedShouldNotifyCtscAdminAndLAWhenRepresentativesNeedServingByPostAndC
         verify(notificationClient).sendEmail(
             eq(ORDER_ISSUED_NOTIFICATION_TEMPLATE_FOR_ADMIN),
             eq("FamilyPublicLaw+ctsc@gmail.com"),
-            dataCaptor.capture(),
```
Comment (akomisarek) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/GeneratedOrderControllerSubmittedTest.java:180]: No verify but it's consistent with old.

File: service/src/test/java/uk/gov/hmcts/reform/fpl/utils/matchers/JsonMatcher.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+import static org.mockito.internal.progress.ThreadSafeMockingProgress.mockingProgress;
+import static org.skyscreamer.jsonassert.JSONCompareMode.STRICT;
+
+public class JsonMatcher implements ArgumentMatcher<Object>, ContainsExtraTypeInfo, Serializable {
+
+    private final Map<String, Object> wanted;
+
+    public static Map<String, Object> eqJson(Map<String, Object> value) {
+        mockingProgress().getArgumentMatcherStorage().reportMatcher(new JsonMatcher(value));
```
Comment (akomisarek) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/utils/matchers/JsonMatcher.java:21]: Do we need this? seems like Mockito internals as it's not properly documented.

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ActionCaseManagementOrderControllerSubmittedTest.java [hunk 3]
```diff
@@ -206,13 +199,13 @@ void shouldNotifyCtscAdminWhenRepresentativesServedByPostAndCtscIsEnabled() thro
...
             eq(ADMIN_EMAIL_ADDRESS),
-            dataCaptor.capture(),
+            any(),
             eq(CASE_ID));
 
         verify(notificationClient).sendEmail(
             eq(ORDER_ISSUED_NOTIFICATION_TEMPLATE_FOR_ADMIN),
             eq("FamilyPublicLaw+ctsc@gmail.com"),
-            dataCaptor.capture(),
+            any(),
```
[the remark this answers is not in the document]
Comment (tomaszpowroznik) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ActionCaseManagementOrderControllerSubmittedTest.java:208]: dont want to mix refactoring with test changes, currently there is no verification of captor

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/GeneratedOrderControllerSubmittedTest.java (hunk 1 above)
Comment (tomaszpowroznik) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/GeneratedOrderControllerSubmittedTest.java:180]: same

File: service/src/test/java/uk/gov/hmcts/reform/fpl/utils/matchers/JsonMatcher.java (hunk 2 above)
Comment (tomaszpowroznik) [reply service/src/test/java/uk/gov/hmcts/reform/fpl/utils/matchers/JsonMatcher.java]: done

**Pull request merged**